### PR TITLE
add agreement parameters in all relevant methods

### PR DESCRIPTION
**Pull request by maf-embraceit**

### CODE REVIEW

File: AltaPayApi/AltaPayApi/Service/Requests/Common/BasePaymentRequestRequest.cs [hunk 1]
```diff
@@ -20,6 +20,13 @@ protected BasePaymentRequestRequest()
 		public string Language { get; set; }
 		public IDictionary<string,object> PaymentInfos { get; set; }
 		public AuthType Type { get; set; }
+		public AgreementType? AgreementType { get; set; }
+		public string AgreementId { get; set; }
+		public AgreementUnscheduledType? AgreementUnscheduledType { get; set; }
+		public string AgreementExpiry { get; set; }
+		public string AgreementFrequency { get; set; }
+		public string AgreementNextChargeDate { get; set; }
+		public string AgreementAdminUrl { get; set; }
```
Comment (emicha) [inline AltaPayApi/AltaPayApi/Service/Requests/Common/BasePaymentRequestRequest.cs:29]: Why not using a complex object, like the PaymentRequestConfig

File: AltaPayApi/AltaPayApi/Service/Requests/ReserveRequest.cs [hunk 2]
```diff
@@ -10,6 +10,13 @@ public class ReserveRequest
 		public string ShopOrderId {get; set;}
 		public Amount Amount { get; set; }
 		public AuthType PaymentType { get; set; }
+		public AgreementType? AgreementType { get; set; }
+		public string AgreementId { get; set; }
+		public AgreementUnscheduledType? AgreementUnscheduledType { get; set; }
+		public string AgreementExpiry { get; set; }
+		public string AgreementFrequency { get; set; }
+		public string AgreementNextChargeDate { get; set; }
+		public string AgreementAdminUrl { get; set; }
```
Comment (emicha) [inline AltaPayApi/AltaPayApi/Service/Requests/ReserveRequest.cs:19]: Why not using a complex object, like the PaymentRequestConfig

**Comment by emicha** [comment]
Missing release notes.

File: AltaPayApi/AltaPayApi/Service/Requests/Common/BasePaymentRequestRequest.cs (hunk 1 above)
Comment (maf-embraceit) [reply AltaPayApi/AltaPayApi/Service/Requests/Common/BasePaymentRequestRequest.cs]: Great idea, done, thank you.

File: AltaPayApi/AltaPayApi/Service/Requests/ReserveRequest.cs (hunk 2 above)
Comment (maf-embraceit) [reply AltaPayApi/AltaPayApi/Service/Requests/ReserveRequest.cs]: Great idea, done, thank you.

**Review by emicha — APPROVED** [review]

**Pull request merged**